### PR TITLE
[R] Fix code section comments

### PR DESCRIPTION
**Pull request by deathaxe**
Enforce whitespace around code section titles like Markdown.

fixes: https://forum.sublimetext.com/t/strange-syntax-highlighting-for-r-comments/63227

see: https://support.rstudio.com/hc/en-us/articles/200484568-Code-Folding-and-Sections-in-the-RStudio-IDE

### CODE REVIEW

**Review by jrappen — APPROVED** [review]

File: R/syntax_test_r.R [hunk 1]
```diff
@@ -40,6 +40,14 @@
 #    ^^^^^^^^^^^^^ entity.name.section.r
 #                 ^^^^^^ comment.line.number-sign.r - entity
 
+######
+# ^^^^^ comment.line.number-sign.r - entity
+
+# # ####
```
Comment (michaelblyons) [inline R/syntax_test_r.R:46]: Add a `- entity` to the `####` at the end, maybe? I know it's covered in other tests further down, but this is the edge case.

**Review by michaelblyons — APPROVED** [review]

**Review by jrappen — APPROVED** [review]

**Review by keith-hall — APPROVED** [review]
